Include main content area

sukiletxe opened the pull request:
Add `<mainn>`in `base.html`, add an empty block `maincontent`inside and
replace `content` with `maincontent`. This allows other templates to
include their data in this area, enhancing accessibility.

Maybe `{{super()}}` could be kept, but I'm not sure, so I removed it. It wouldn't serve any purpose, anyway.

sukiletxe committed:
Include main content area. Fixes #14.

Add a `<main>` element in `base.html`, and an empty block `maincontent`
inside it. Replace `content` in other templates with `maincontent`. This
allows other templates to include their data in this area, enhancing
accessibility.

mitchtbaum commented:
Looks good.  Some simple styling changes: commit message `<main>` spelling and `base.html` `maincontent` and `endblock` jinja tags' wrapper spaces.

When I last worked on this, the `` {{ super() }}` blocks seemed necessary for pulling in the parent template's contents, but I would like to understand why it doesn't serve any purpose before committing.  I based that use on [Flask's template inheritance docs](http://flask.pocoo.org/docs/0.10/patterns/templateinheritance/) and [Jinja's template inheritance docs](http://jinja.pocoo.org/docs/dev/templates/#template-inheritance)), but now that I tested your changes it seems to work fine without it.  Please put that change in another PR to review it separately and to add a commit message that details why we changed it.

sukiletxe commented:
OK, I'll amend that to change the commit message.

I omitted `{{super}}()` because there is no content in ` maincontent `, 
it gets completely populated by the article, page, tags list, etc. Ican 
put it again, but it won't make any difference (if it can be aplied to 
empty blocks, which I don't know).

sukiletxe commented:
I've implemented the changes you requested. I've also added `{{ super() 
}}` to the templates again. But, as I said, it won't make any 
difference, as the block is completely empty in the beginning.

El 23/02/2016 a las 6:24, Mitchell Tannenbaum escribió:
>
> Looks good. Some simple styling changes: commit message |<main>| 
> spelling and |base.html| |maincontent| and |endblock| jinja tags' 
> wrapper spaces.
>
> When I last worked on this, the `|{{ super() }}| blocks seemed 
> necessary for pulling in the parent template's contents, but I would 
> like to understand why it doesn't serve any purpose before committing. 
> I based that use on Flask's template inheritance docs 
> <http://flask.pocoo.org/docs/0.10/patterns/templateinheritance/> and 
> Jinja's template inheritance docs 
> <http://jinja.pocoo.org/docs/dev/templates/#template-inheritance>), 
> but now that I tested your changes it seems to work fine without it. 
> Please put that change in another PR to review it separately and to 
> add a commit message that details why we changed it.
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/getpelican-themes/salutations/pull/15#issuecomment-187543703>.
>
